identity-account/src/stronghold/error.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use iota_stronghold::procedures::ProcedureError;
+
+pub type IotaStrongholdResult<T> = Result<T, StrongholdError>;
+
+#[derive(Debug, thiserror::Error, strum::IntoStaticStr)]
+pub enum StrongholdError {
+  /// Caused by errors from the [iota_stronghold] crate.
+  #[error(transparent)]
+  StrongholdActorError(#[from] iota_stronghold::ActorError),
+  #[error(transparent)]
+  StrongholdWriteError(#[from] iota_stronghold::WriteError),
+  #[error(transparent)]
+  StrongholdReadError(#[from] iota_stronghold::ReadError),
+  #[error(transparent)]
+  StrongholdFatalEngineError(#[from] iota_stronghold::FatalEngineError),
+  #[error(transparent)]
+  StrongholdMailboxError(#[from] iota_stronghold::MailboxError),
+  /// Caused by attempting to perform an invalid IO operation.
+  #[error(transparent)]
+  IoError(#[from] std::io::Error),
+  /// Caused by receiving an unexpected return value from a Stronghold procedure.
+  #[error("Stronghold procedure returned unexpected type")]
+  StrongholdProcedureFailure(#[from] ProcedureError),
+  /// Caused by attempting to access a Stronghold snapshot without a password.
+  #[error("Stronghold snapshot password not found")]
+  StrongholdPasswordNotSet,
+  /// Caused by an internal panic in the Stronghold runtime.
+  #[error("Stronghold mutex poisoned: {0}")]
+  StrongholdMutexPoisoned(&'static str),
+  /// Caused by errors from an invalid Stronghold procedure.
+  #[error("Stronghold error: {0}")]
+  StrongholdResult(String),
+  #[error("Record Error")]
+  RecordError,
```

Comment:
I removed `Records` for now.

Reply:
> Do we not need `records` at all, even to implement listing all DIDs in a `Stronghold` #567 ?

I guess we could use it for that, yeah. I actually thought this was wrapping the stronghold record hints API which is going to be removed with the refactor, which is why I didn't look that closely at what `Records` was doing. I think we can also just reintroduce it in a future PR, if it turns out to be useful then, since it's already removed now.

Reply:
I reverted the commit that removed the records 😅